docs/proposals/query-blocking.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "XXXX: Query blocking"
+description: "Query blocking"
+draft: true
+---
+
+# XXXX: Query blocking
+
+**Author:** @wilfriedroset
+
+**Date:** 05/2023
+
+**Sponsor(s):** @username of maintainer(s) willing to shepherd this MID
+
+**Type:** Feature
+
+**Status:** Draft
+
+**Related issues/PRs:**
+
+---
+
+## Background
+
+The `query-frontend` is a component that is responsible for query acceleration through splitting, sharding, caching, and other techniques. It is also responsible for enforcing limits on queries such as total query time range.
+
+## Problem Statement
+
+In certain situations, you may not be able to control the queries being sent to your Mimir installation. These queries may be intentionally or unintentionally expensive to run, and they may affect the overall stability or cost of running your service.
+
+Example:
+
+```
+sum(increase(tfe_run_current_count{status=~"applied|planned_and_finished", instance="host1", organization_name=~".*", workspace_name=~".*"}[2674800s]) > 0)
+```
+
+The query above can be harmful as the wildcard regex would match any label values including empty string. The result of such query is not cached by Mimir. As such the read path would be under pressure if the query is repeated by the tenant (e.g: dashboard in auto repeat).
+
+## Goals
+
+- Prevent harmful queries to be processed by Mimir's read path.
+
+## Non-Goals (optional)
+
+- Rewrite queries in a non harmful way.
+- Optimize queries on the fly.
+- Blocked remote read.
+- Block similar queries with a single definition. Blocking `metric_name{label1=value1, label2=value2}` does not automatically block `metric_name{label2=value2, label1=value1}`.
```

Comment:
I wonder if this shouldn't be the goal to some extent. I think avoiding whitespace changes at least makes sense. At the same time, I understand if as a first iteration you don't want to include this "normalization"

Reply:
Taken into account whitespace changes seems easy but I would rather either completely normalize the queries or not at all. I reckon it is simpler to understand from an operator point of view.